pkg/operator/ceph/csi/spec.go

```diff
@@ -227,7 +227,7 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	csiPluginImage := strings.Split(CSIParam.CSIPluginImage, ":")
 	// as ceph-csi v2.x.x support resizer, enable it
 	if len(csiPluginImage) > 1 {
-		if strings.HasPrefix(csiPluginImage[1], "v2.") {
+		if strings.HasPrefix(csiPluginImage[len(csiPluginImage)-1], "v2.") {
```

Comment:
Questions:
- Are there other places we need this change? For example, above on line 222.
- Checking for an image tag isn't always reliable. With #4256 we detect the actual version of the csi driver. Can we use this instead? I'm fine if it's a separate PR though since it's a bigger change.

Reply:
updated check for the attacher as well. if you are okay to start both attacher and cephcsi container to check the built version we can do it, it may slow down starting csi driver for the first time.
even if someone builds a custom image with the different version we may fail.